### PR TITLE
add agents property to all spaces

### PR DESCRIPTION
**Pull request by quaquel**
This adds an `agents` property to all spaces which returns an AgentSet containing the agents in the space. See #2416 for background discussion

**TODO**
- [x] tests

### CODE REVIEW

File: mesa/space.py [hunk 1]
```diff
@@ -1333,6 +1353,19 @@ def __init__(
...
         self._agent_to_index: dict[Agent, int | None] = {}
 
+    @property
+    def agents(self) -> AgentSet:
+        """Return an AgentSet with the agents in the space"""
+        agents = [agent for agent in self._agent_to_index]
+
+        # getting the rng is a bit hacky because old style spaces don't have the rng
+        try:
+            rng = agents[0].random
```
Comment (EwoutH) [inline mesa/space.py:1363]: Do we know the concept of an empty AgentSet in Mesa?

Comment (quaquel) [reply mesa/space.py:1363]: yes, agentsets can be empty

Comment (EwoutH) [reply mesa/space.py:1363]: Then it should return an empty AgentSet, not an IndexError. Maybe pass `model.random`?

Comment (quaquel) [reply mesa/space.py:1363]: 1. old style spaces don't have a reference to the model or to a random number generator
2. Please look at the code: I am returning an empty set. This you are commenting on is only for getting the random number generator. 

```python
try:
    rng = agents[0].random
except IndexError:
    # there are no agents in the space
    rng = None
return AgentSet(agents, random=rng)


```

Comment (EwoutH) [reply mesa/space.py:1363]: Sorry, you’re right, I read it wrong.

Thanks for explaining!

**Review by EwoutH — APPROVED** [review]
Looks good, thanks!

**Pull request merged**